### PR TITLE
Feature/call automation implement idempotency

### PR DESCRIPTION
**Pull request by yyw-msft**
- added idempotency headers for: 
  answer call
  create call
  reject call
  redirect call 
  terminate
  transfer to participant
  add participants
  remove participants

- added RepeatabilityHeaders common class with unit tests
- added CallAutomationErrorMessages class in Misc
- added recording test for CreateCall

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs [hunk 1]
```diff
@@ -109,12 +108,23 @@ protected CallAutomationClient()
...
         /// <param name="cancellationToken"> The cancellation token. </param>
         /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
         /// <exception cref="ArgumentNullException"><paramref name="incomingCallContext"/> is null.</exception>
-        public virtual async Task<Response<AnswerCallResult>> AnswerCallAsync(string incomingCallContext, Uri callbackUri, CancellationToken cancellationToken = default)
+        /// <exception cref="ArgumentException"><paramref name="repeatabilityRequestId"/> <paramref name="repeatabilityFirstSent"/> Repeatability headers are set incorrectly.</exception>
+        public virtual async Task<Response<AnswerCallResult>> AnswerCallAsync(string incomingCallContext, Uri callbackUri, Guid? repeatabilityRequestId = null, String repeatabilityFirstSent = default, CancellationToken cancellationToken = default)
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs:117]: Question: If repeatability variables are not set, shouldn't this be set by us?

Comment (yyw-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationClient.cs:117]: my understanding is that repeatability variables are completely optional. Contoso can either set them to take advantage of the feature or ignore them so every request is a new request. @fangchen0601 just to confirm? If repeatability variables are not set, then sdk will not add the headers to request. eg. https://github.com/Azure/azure-sdk-for-net/blob/9b44543bc04aa4f3deb62ec37795bc7fddceafef/sdk/communication/Azure.Communication.CallAutomation/src/Generated/ServerCallingRestClient.cs#L51

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/AddParticipantsOptions.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    /// </summary>
+    public class AddParticipantsOptions: RepeatabilityHeaders
+    {
+        /// <summary>
+        /// Creates a new AddParticipantsOptions object.
+        /// </summary>
+        /// <param name="participantsToAdd"></param>
+        /// <param name="repeatabilityRequestId"></param>
+        /// <param name="repeatablityFirstSent"></param>
+        public AddParticipantsOptions(IEnumerable<CommunicationIdentifier> participantsToAdd, Guid? repeatabilityRequestId = null, string repeatablityFirstSent = default)
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/AddParticipantsOptions.cs:20]: Optional values to not be included in the constructor, to keep it consistent throughout the sdk.

... required service method parameters are modeled as Options class constructor parameters and get-only properties, and optional parameters are get-set properties.

Comment (yyw-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/AddParticipantsOptions.cs]: updated. ty!

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/TransferToParticipantOptions.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    /// </summary>
+    public class TransferToParticipantOptions: RepeatabilityHeaders
+    {
+        /// <summary>
+        /// Creates a new TransferToParticipantOptions object.
+        /// </summary>
+        /// <param name="targetParticipant"></param>
+        /// <param name="repeatabilityRequestId"></param>
+        /// <param name="repeatablityFirstSent"></param>
+        public TransferToParticipantOptions(CommunicationIdentifier targetParticipant, Guid? repeatabilityRequestId = null, string repeatablityFirstSent = default)
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/TransferToParticipantOptions.cs:19]: Needs update here also for options in constructor

Comment (yyw-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/TransferToParticipantOptions.cs]: ty for catching! updated!

**Review by minwoolee-msft — APPROVED** [review]

**Pull request merged**